sheetScript.gs

```diff
@@ -463,50 +449,34 @@ function deletePlaylistItems(playlistId, deleteBeforeTimestamp) {
   }
 }
 
-function getAllChannelIds_OLD() { // Note: this function is not used.
-  var channelIds = [];
+//
+// Functions for Housekeeping
+// Makes Web App, function call from Google Sheets, add errors, etc
+//
 
-  // First call
-  try {
-    var results = YouTube.Subscriptions.list('snippet', {
-      mine: true,
-      maxResults: 50
-    });
-  } catch (e) {
-    Logger.log("ERROR: " + "Message: [" + e.message + "] Details: " + JSON.stringify(e.details));
-    return;
-  }
-  for (var i = 0; i < results.items.length; i++) {
-    var item = results.items[i];
-    channelIds.push(item.snippet.resourceId.channelId);
+function playlist(pl, sheetID){
+  var sheet = SpreadsheetApp.openById(sheetID).getSheets()[0];
+  var data = sheet.getDataRange().getValues();
+  if (pl == undefined){
+    pl = reservedTableRows;
+  } else {
+    pl = Number(pl) + reservedTableRows - 1;  // I like to think of the first playlist as being number 1.
   }
-
-  // Other calls
-  var nextPageToken = results.nextPageToken;
-  for (var pageNo = 0; pageNo < (-1+Math.ceil(results.pageInfo.totalResults / 50.0)); pageNo++) {
-
-    try {
-      results = YouTube.Subscriptions.list('snippet', {
-        mine: true,
-        maxResults: 50,
-        pageToken: nextPageToken
-      });
-    } catch (e) {
-      Logger.log("ERROR: " + "Message: [" + e.message + "] Details: " + JSON.stringify(e.details));
-      continue;
-    }
-    for (var i = 0; i < results.items.length; i++) {
-      var item = results.items[i];
-      channelIds.push(item.snippet.resourceId.channelId);
-    }
-
-    nextPageToken = results.nextPageToken;
+  if (pl > sheet.getLastRow()){
+    pl = sheet.getLastRow();
   }
+  var playlistId = data[pl][reservedColumnPlaylist];
+  return playlistId
+}
 
-  Logger.log('Acquired subscriptions %s, Actual subscriptions %s', channelIds.length, results.pageInfo.totalResults);
-  return channelIds;
+// Log errors in debug sheet and throw an error
+function addError(s) {
+  Logger.log(s);
+  errorflag = true;
+  plErrorCount += 1;
 }
 
+// Function to Set Up Google Spreadsheet
 function onOpen() {
   SpreadsheetApp.getActiveSpreadsheet().addMenu("Functions", [{name: "Update Playlists", functionName: "insideUpdate"}]);
```

Comment:
We can remove the `insideUpdate` function (here and its implementation) since `updatePlaylists` can pull correct sheet id itself.
```suggestion
  SpreadsheetApp.getActiveSpreadsheet().addMenu("Functions", [{name: "Update Playlists", functionName: "updatePlaylists"}]);
```

Reply:
We can also remove the implementation of the `insideUpdate` function.

Reply:
I'm sorry something funny is happening with my git. I thought I had removed that one too.